gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -404,6 +405,36 @@ void testTrinoQueryPropertiesTableExtraction(String query, Set<String> catalogs,
         assertThat(trinoQueryProperties.getCatalogs()).isEqualTo(catalogs);
     }
 
+    @Test
+    void testWithQueryNameExcluded()
+            throws IOException
+    {
+        String query = """
+                WITH dos AS (SELECT c1 from cat.schem.tbl1),
+                uno as (SELECT c1 FROM dos)
+                SELECT c1 FROM uno, dos
+                """;
+        BufferedReader bufferedReader = new BufferedReader(new StringReader(query));
```

Comment:
nit: No need to reuse `bufferedReader`. I would inline this variable.

Reply:
this is reused in line 431 so, and i think both is needed as we have two TrinoQueryProperties objects

Reply:
I know it's reused in 431. The question is why do we need to reuse the object?

Reply:
i don't quite understand your point. do you meant by changing to this?

```java
when(mockRequestWithDefaults.getReader()).thenReturn(new BufferedReader(new StringReader(query)));
when(mockRequestNoDefaults.getReader()).thenReturn(new BufferedReader(new StringReader(query)));

Reply:
Yes.

Reply:
ok :) thx for feedbacks. Learning alot